src/analysis/complex/upper_half_plane/basic.lean

```diff
@@ -65,7 +65,9 @@ def mk (z : ℂ) (h : 0 < z.im) : ℍ := ⟨z, h⟩
 @[simp] lemma coe_re (z : ℍ) : (z : ℂ).re = z.re := rfl
 
 @[simp] lemma mk_re (z : ℂ) (h : 0 < z.im) : (mk z h).re = z.re := rfl
+@[simp] lemma mk_re' (z : ℂ) (h) : re (⟨z, h⟩ : ℍ) = z.re := rfl
```

Comment:
Which `re` is this on the LHS? (I'll check when I'm next at lean if you're not)

Reply:
This is `def re (z : ℍ) := (z : ℂ).re` at line 57.

Reply:
Then isn't this exactly the same as the lemma above?

Reply:
Ah, I see: these are about `subtype.mk` not `mk`. I still think we probably don't want these lemmas; do you actually use them in this PR?

Reply:
Yes I think `mk_re'` is used in `simp` somewhere.

Reply:
Can you remove it anyway? If it's being used by simp, then you have a proof somewhere that is unfolding things it shouldn't

Reply:
These were used in a previous version of some proof(s) which unfolded all the way to `ℂ` instead of staying in `ℍ`. 

Personally I think it's fine to keep them but I I've pushed a commit which drops them since they are not in fact needed any more.